lib/reek/examiner.rb

```diff
@@ -97,12 +97,15 @@ def smelly?
     #
     # :reek:TooManyStatements { max_statements: 8 }
     def run
-      return [] unless syntax_tree
-      begin
-        examine_tree
-      rescue Errors::BaseError => exception
-        raise unless @error_handler.handle exception
-        []
+      if source.valid_syntax? && syntax_tree
```

Comment:
I did check our specs and we have at least [one example](https://github.com/troessner/reek/blob/master/spec/reek/source/source_code_spec.rb#L20) that does check this indeed. I just check out @pocke branch and ran it on both empty and comment-only source files and it worked just. I assume parser's behaviour now has changed to always return a syntax tree.

So seems like we're good with this change ;)